packages/core/lib/commands/db/commands/serve.js

```diff
@@ -16,13 +16,17 @@ const command = {
    */
   run: async function (argv) {
     const Config = require("@truffle/config");
-    const {serve} = require("@truffle/db");
+    const { serve } = require("@truffle/db");
 
     const config = Config.detect(argv);
     const port = (config.db && config.db.port) || 4444;
     const host = (config.db && config.db.host) || "127.0.0.1";
+    const serveOptions =
+      config.db && config.db.directory
+        ? { directory: config.db.directory }
```

Comment:
Let's see if we can try to remove the need for this, favoring instead (e.g.):

```javascript
module.exports = {
  // rest of truffle-config.js
  db: {
    adapter: {
      name: "sqlite",
      settings: {
        directory: "..."
      }
    }
  }
}
```

This way, there's no need to pass a directory if the adapter won't need it (like couch/memory). It'd mean there's the loss of a convenient way to just say "use this base directory", but I'm not sure the need to make that convenient? It's not that much extra work to add a couple layers of wrapping for something for something that I'm assuming won't be used much.

Anyway not sure if there's a clean way to get rid of this but I'll write thoughts about it as I scroll down.

Reply:
This doesn't require that the user pass a directory, it will just use the directory that is supplied if it is present. However, if you think it is better to namespace it like this then I have no problem with that. Although I suppose the directory is not adapter-specific so maybe it doesn't belong in `adapter`. What do you think?

Reply:
My thinking that it's adapter-specific is because that field only ends up getting used by two out of the four adapters

Reply:
Ok, that makes sense